tensorflow/core/kernels/conv_ops_benchmark_test.cc

```diff
@@ -97,19 +95,45 @@ static Conv2DGraph Conv2D(int batch, int height, int width, int in_depth,
   Node* images = test::graph::Constant(graph, images_t, "images");
   Node* filter = test::graph::Constant(graph, filter_t, "filter");
 
+  graph->AddEdge(graph->source_node(), 0, images, 0);
+  graph->AddEdge(graph->source_node(), 1, filter, 0);
+
+  // Add shape sizes to images and filter.
+  AttrValue attr_input_shape;
+  TensorShapeProto* proto = attr_input_shape.mutable_list()->add_shape();
+  proto->add_dim()->set_size(batch);
+  proto->add_dim()->set_size(height);
+  proto->add_dim()->set_size(width);
+  proto->add_dim()->set_size(in_depth);
+  proto = attr_input_shape.mutable_list()->add_shape();
+  proto->add_dim()->set_size(filter_h);
+  proto->add_dim()->set_size(filter_w);
+  proto->add_dim()->set_size(in_depth);
+  proto->add_dim()->set_size(out_depth);
+
   Node* conv2d;
-  auto builder = IsMKLEnabled()
-                     ? NodeBuilder(graph->NewName("conv"), "_MklNativeConv2D")
-                           .Attr("_kernel", MKL_OP_LABEL)
-                     : NodeBuilder(graph->NewName("conv"), "Conv2D");
+  auto builder = NodeBuilder(graph->NewName("conv"), "Conv2D");
   TF_CHECK_OK(builder.Input(images)
                   .Input(filter)
                   .Attr("T", DataTypeToEnum<T>::value)
                   .Attr("strides", {1, 1, 1, 1})
                   .Attr("padding", "SAME")
                   .Attr("data_format", ToString(data_format))
+                  .Attr("_input_shapes", attr_input_shape)
                   .Finalize(graph, &conv2d));
 
+  int conv2d_node_id = conv2d->id();
+  if (IsMKLEnabled()) {
+    std::unique_ptr<Graph>* ug = new std::unique_ptr<Graph>(graph);
+    RunMklLayoutRewritePass(ug);
```

Comment:
This needs to be guarded by `INTEL_MKL`, otherwise we see build failures.

Reply:
Thanks. Fixed in 4433d4b